lualib/resty/kong/tls.lua

```diff
@@ -131,4 +154,38 @@ do
 end
 
 
+do
+    local ALLOWED_PHASES = {
+        ['rewrite'] = true,
+        ['balancer'] = true,
+        ['access'] = true,
+    }
+
+    function _M.set_upstream_cert_and_key(chain, key)
+        if not ALLOWED_PHASES[get_phase()] then
+            error("API disabled in the current context", 2)
+        end
+
+        if not chain or not key then
+            error("chain and key must not be nil")
+        end
+
+        local r = get_request()
+        size_ptr[0] = DEFAULT_CERT_CHAIN_SIZE
```

Comment:
It doesn't look like we need this allocation?

Reply:
Copy-n-paste error. Thanks for pointing out!